cmd/limactl/start.go

```diff
@@ -49,7 +53,22 @@ func loadOrCreateInstance(cmd *cobra.Command, args []string) (*store.Instance, e
 		err      error
 	)
 
-	if argSeemsYAMLPath(arg) {
+	if argSeemsHTTPURL(arg) {
+		instName, err = instNameFromHTTPURL(arg)
+		if err != nil {
+			return nil, err
+		}
+		logrus.Debugf("interpreting argument %q as a http url for instance %q", arg, instName)
+		resp, err := http.Get(arg)
+		if err != nil {
+			return nil, err
+		}
+		defer resp.Body.Close()
+		yBytes, err = io.ReadAll(resp.Body)
```

Comment:
Should check body size before calling ReadAll

Reply:
Sorry for jumping the gun on merging already 😞

Reply:
Good call, hopefully it would return an error in that case ?

Reply:
Just used the https://pkg.go.dev/net/http sample code

Reply:
We can use https://pkg.go.dev/io#LimitedReader and it raises io.EOF when it reaches the limit (e.g., 1MB).
We should wrap the io.EOF error with a human-readable text.

Reply:
ah, so it was limit "protection" rather than looking for zero bytes

Reply:
but wouldn't the same thing happen with a local file as well ?

Reply:
PR: https://github.com/lima-vm/lima/pull/279